src/median.js

```diff
@@ -1,5 +1,31 @@
-export function median(data) {
+/* 
+ * Returns the meadian x value from a histogram of sorted abscissa
+ * @param {object} data - x sorted abscissa by incresing order and y the number of occurrence of each x value
+ * @returns {number} xmedian - the median of x values 
+ */
+
+ export function median(data) {
   const { x, y } = data;
 
-  return 0;
+  if (x.length === 1) {
+    return x[0];
+  }
+
+  let sumY = 0;
+  let cumSumY = new Array(y.length);
+
+  for (let i = 0; i < y.length; i++) {
+    sumY += y[i];
+    cumSumY[i] = sumY;
+  }
+
+  for (let k = 0; k < cumSumY.length; k++) {
```

Comment:
You may reuse the variable `i`  The previous `i` variable is scoped in the previous loop